### PR TITLE
Switch to use MSWebBrowser as default lib display

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Switch to use MSWebBrowser as default:
1. LibraryViewExtensionMSWebBrowser will build to Dynamo build folder
2. LIbraryViewExtension will build to its own bin folder
3. When launching Dynamo, Dynamo will load LibraryViewExtensionMSWebBrowser as default lib view extension
4. Dynamo folder will be missing from a few files including all the CEFSharp dependencies and locales

What it looks like, notice the scroll bar
image

What the build contains for `LibraryViewExtension` which is no longer part of DynamoCore:
image

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj [hunk 1]
```diff
@@ -24,7 +24,7 @@
     <DebugSymbols>true</DebugSymbols>
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
-    <OutputPath>bin\Debug\</OutputPath>
+    <OutputPath>..\..\bin\AnyCPU\Debug\</OutputPath>
```
Comment (mjkkirschner) [inline src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj:27]: I don't think this should usually be set directly like this, instead import our normal configuration like in other csproj files and delete these lines, then the output path will be set correctly for each config without needing to update it.

```
  <Import Project="$(SolutionDir)/Config/CS.props" />
```

Comment (mjkkirschner) [reply src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj:27]: for example, your release path is still wrong in this commit.

**Comment by mjkkirschner** [comment]
are you concerned about the missing files for patch installers? Will you be manually copying them back in after build?

**Comment by QilongTang** [comment]
> are you concerned about the missing files for patch installers? Will you be manually copying them back in after build?

Since this would be for Global launch, I am not worries about the missing files for patch installers yet. I do remember Civil3D may back port updated Dynamo there but since they were already using `MSWebBrowser`, seems no issue should be raised on that end?

File: src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj (hunk 1 above)
Comment (QilongTang) [reply src/LibraryViewExtensionMSWebBrowser/LibraryViewExtensionMSWebBrowser.csproj:27]: Thanks, will correct this

**Comment by mjkkirschner** [comment]
@QilongTang I'm not sure why I didn't do this in the past - maybe there is a reason - but can you checkout why layoutspec.json is duplicated in both projects? It would be nice if one was a link to the other so we only had to update it once in one spot when adding new nodes - or do you intend to drop the CEF project completely?

I mean adding existing item to project, and then using `add file link`:

**Comment by QilongTang** [comment]
> @QilongTang I'm not sure why I didn't do this in the past - maybe there is a reason - but can you checkout why layoutspec.json is duplicated in both projects? It would be nice if one was a link to the other so we only had to update it once in one spot when adding new nodes - or do you intend to drop the CEF project completely?
> 
> I mean adding existing item to project, and then using `add file link`:

@mjkkirschner I found that using linked layoutSpec will break the library display

**Comment by mjkkirschner** [comment]
> > @QilongTang I'm not sure why I didn't do this in the past - maybe there is a reason - but can you checkout why layoutspec.json is duplicated in both projects? It would be nice if one was a link to the other so we only had to update it once in one spot when adding new nodes - or do you intend to drop the CEF project completely?
> > I mean adding existing item to project, and then using `add file link`:
> 
> @mjkkirschner I found that using linked layoutSpec will break the library display so I left it untouched

yeah - odd, the file needs to be embedded into the binary for it to be found at runtime, but maybe that doesn't make sense with a linked file - anyhow lets worry about it later - eventually we'll only have one of these projects anyway.

**Comment by mjkkirschner** [comment]
@QilongTang I'm confused - in your image above you show a bunch of CEF dependencies in the bin folder, I thought you were getting rid of them as per this comment: `Dynamo folder will be missing from a few files including all the CEFSharp dependencies and locales`

File: src/LibraryViewExtension/web/library/layoutSpecs.json [hunk 2]
```diff
@@ -5,13 +5,13 @@
       "iconUrl": "",
       "elementType": "section",
       "showHeader": false,
-      "include": [ ],
+      "include": [],
```
Comment (mjkkirschner) [inline src/LibraryViewExtension/web/library/layoutSpecs.json:8]: what are these changes about?

**Comment by QilongTang** [comment]
> @QilongTang I'm confused - in your image above you show a bunch of CEF dependencies in the bin folder, I thought you were getting rid of them as per this comment: `Dynamo folder will be missing from a few files including all the CEFSharp dependencies and locales`

Yeah, please check the path, these are under LibraryViewExtension sub folder for future nuget wrapping

File: src/LibraryViewExtensionMSWebBrowser/ViewExtension.cs [hunk 3]
```diff
@@ -26,7 +26,7 @@ public string UniqueId
             get { return "63cd0755-4a36-4670-ae89-b68e772633c4"; }
         }
 
-        public static readonly string ExtensionName = "LibraryUI2";
+        public static readonly string ExtensionName = "LibraryUI - MSWebBrowser";
```
Comment (mjkkirschner) [inline src/LibraryViewExtensionMSWebBrowser/ViewExtension.cs:29]: my reasoning for this funny naming was to avoid exposing the implementation in the name - not a big deal either way.

File: src/LibraryViewExtension/web/library/layoutSpecs.json (hunk 2 above)
Comment (QilongTang) [reply src/LibraryViewExtension/web/library/layoutSpecs.json:8]: Visual Studio Code Json formatting

**Review by mjkkirschner — APPROVED** [review]
looks good - just a few questions.

File: src/LibraryViewExtensionMSWebBrowser/ViewExtension.cs (hunk 3 above)
Comment (QilongTang) [reply src/LibraryViewExtensionMSWebBrowser/ViewExtension.cs:29]: I can revert this or do you have an alternative naming option?

Comment (mjkkirschner) [reply src/LibraryViewExtensionMSWebBrowser/ViewExtension.cs:29]: nah, it's fine - lgtm.

**Comment by QilongTang** [comment]
There are lots of regressions, investigating

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 4]
```diff
@@ -45,7 +45,7 @@
 // to distinguish one build from another. AssemblyFileVersion is specified
 // in AssemblyVersionInfo.cs so that it can be easily incremented by the
 // automated build process.
-[assembly: AssemblyVersion("2.13.0.2104")]
+[assembly: AssemblyVersion("2.13.0.2344")]
```
Comment (reddyashish) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs:48]: Commenting just to make sure, you revert the changes to this file.

Comment (QilongTang) [reply src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs]: Thanks, reverted!

**Comment by QilongTang** [comment]
The previous failures all pass locally for me. Merging.
